runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -200,6 +200,39 @@ ClientSessionData::processUnloadedClasses(const std::vector<TR_OpaqueClassBlock*
       writeReleaseClassUnloadRWMutex();
    }
 
+void
+ClientSessionData::processIllegalFinalFieldModificationList(const std::vector<TR_OpaqueClassBlock*> &classes)
+   {
+   const size_t numOfClasses = classes.size();
+
+   if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+      TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+         "compThreadID=%d will process a list of %zu classes with illegal final field modification for clientUID %llu",
+            TR::compInfoPT->getCompThreadId(), numOfClasses, (unsigned long long)_clientUID);
+
+   if (numOfClasses > 0)
+      writeAcquireClassUnloadRWMutex(); //TODO: use RAII style to avoid problems with exceptions
+
+      {
+      OMR::CriticalSection processClassesWithIllegalModification(getROMMapMonitor());
+      for (auto clazz : classes)
+         {
+         auto it = _romClassMap.find((J9Class*)clazz);
+         if (it != _romClassMap.end())
+            {
+            it->second._classFlags |= J9ClassHasIllegalFinalFieldModifications;
+            if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+               TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+                  "compThreadID=%d found clazz %p in the cache and updated bit J9ClassHasIllegalFinalFieldModifications to 1\n",
```

Comment:
`TR::compInfoPT->getCompThreadId()` uses thread local storage which has overhead. Let's cache this and not call it repeatedly in the loop.